extensions/wikia/DefaultMessages/DefaultMessagesCache.php

```diff
@@ -0,0 +1,521 @@
+		$where = array(); # Debug info, delayed to avoid spamming debug log too much
+		$cacheKey = $this->memcKey( 'messages', $code ); # Key in memc for messages
+
+
+		# (1) local cache
+		# Hash of the contents is stored in memcache, to detect if local cache goes
+		# out of date (due to update in other thread?)
+		if ( $this->mLocalMessageCache !== false ) {
+			wfProfileIn( __METHOD__ . '-fromlocal' );
+
+			$hash = $this->mMemc->get( $this->memcKey( 'messages', $code, 'hash' ) );
+			if ( $hash ) {
+				$success = $this->loadFromLocal( $hash, $code );
+				if ( $success ) $where[] = 'got from local cache';
+			}
+			wfProfileOut( __METHOD__ . '-fromlocal' );
+		}
+
+		# (2) memcache
+		# Fails if nothing in cache, or in the wrong version.
+		if ( !$success ) {
+			wfProfileIn( __METHOD__ . '-fromcache' );
+			$cache = $this->mMemc->get( $cacheKey );
+			$success = $this->setCache( $cache, $code );
+			if ( $success ) {
+				$where[] = 'got from global cache';
+				$this->saveToCaches( $cache, false, $code );
+			}
+			wfProfileOut( __METHOD__ . '-fromcache' );
+		}
+
+
+		# (3)
+		# Nothing in caches... so we need create one and store it in caches
+		if ( !$success && $useDB) {
+			$where[] = 'cache is empty';
+			$where[] = 'loading from database';
+
+			$this->lock($cacheKey);
+
+			$cache = $this->loadFromDB( $code );
+			$success = $this->setCache( $cache, $code );
+			if ( $success ) {
+				$this->saveToCaches( $cache, true, $code );
+			}
+
+			$this->unlock($cacheKey);
+		}
+
+		if ( !$success ) {
+			# Bad luck... this should not happen
+			$where[] = 'loading FAILED - cache is disabled';
+			$info = implode( ', ', $where );
+			wfDebug( __METHOD__ . ": Loading $code... $info\n" );
+			$this->mDisable = true;
+			$this->mCache = false;
+		} else {
+			# All good, just record the success
+			$info = implode( ', ', $where );
+			wfDebug( __METHOD__ . ": Loading $code... $info\n" );
+			$this->mLoadedLanguages[$code] = true;
+		}
+		wfProfileOut( __METHOD__ );
+		return $success;
+	}
+
+	/**
+	 * Loads cacheable messages from the database. Messages bigger than
+	 * $wgMaxMsgCacheEntrySize are assigned a special value, and are loaded
+	 * on-demand from the database later.
+	 *
+	 * @param $code Optional language code, see documenation of load().
+	 * @return Array: Loaded messages for storing in caches.
+	 */
+	function loadFromDB( $code = false ) {
+		wfProfileIn( __METHOD__ );
+		global $wgMaxMsgCacheEntrySize;
+		global $wgDefaultMessagesDB;
+		$dbr = wfGetDB( DB_SLAVE, array(), $wgDefaultMessagesDB );
+		$cache = array();
+
+		# Common conditions
+		$conds = array(
+			'page_is_redirect' => 0,
+			'page_namespace' => NS_MEDIAWIKI,
+		);
+
+		if ( $code ) {
+			# Is this fast enough. Should not matter if the filtering is done in the
+			# database or in code.
+			if ( $code !== 'en' ) {
+				# Messages for particular language
+				$conds[] = "page_title" . $dbr->buildLike( $dbr->anyString(), "/$code" );
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "page_title" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)